app/controllers/application_controller.rb

```diff
@@ -1,5 +1,26 @@
 class ApplicationController < ActionController::Base
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  def current_user
+    current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+  end
+  helper_method :current_user
+
+  def authorized?
+    params[:id].to_i == session[:user_id]
+  end
+
+  def require_authorization
+    render :file => "public/403" unless authorized?
+  end
+
+  def logged_in?
```

Comment:
This is of questionable value since if no-one is logged in, current_user is nil which is falsy.